src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -471,11 +471,21 @@ private void BeforeAttribute()
                 return;
             }
 
+            // Minimized attribute
+            // Need to modify this code to account for space inbetween equals when
+            // https://github.com/aspnet/Razor/issues/123 is completed.
             if (!At(HtmlSymbolType.Equals))
             {
-                // Saw a space or newline after the name, so just skip this attribute and continue around the loop
-                Accept(whitespace);
-                Accept(name);
+                // Output anything prior to the attribute, in most cases this will be the tag name:
+                // |<input| checked />
```

Comment:
confusing update that makes more sense given the previous comment at line 476.  why are you changing the code that parses whitespace before an attribute name?  what do the _two_ vertical bars in your example mean?